### PR TITLE
KV fix: add exercise key to submission inputs

### PR DESCRIPTION
**Pull request by fabiotudone-da**
### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [x] Include appropriate tests
- [x] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [x] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/InputsAndEffects.scala [hunk 1]
```diff
@@ -85,25 +93,25 @@ private[kvutils] object InputsAndEffects {
...
-              inputs += globalKeyToStateKey(
-                GlobalKey(create.coinst.template, forceNoContractIds(keyWithMaintainers.key.value)))
+              addKeyInput(create.coinst.template, keyWithMaintainers)
             }
 
           case exe: Node.NodeExercises.WithTxValue[NodeId, Value.ContractId] =>
             addContractInput(exe.targetCoid)
+            exe.key.foreach { keyWithMaintainers =>
+              addKeyInput(exe.templateId, keyWithMaintainers)
+            }
```
Comment (SamirTalwar-DA) [inline ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/InputsAndEffects.scala:108]: Does this work even if it's a regular consuming exercise on a contract that happens to have a key?

The tests imply as much, but I want to be sure.

File: ledger/participant-state/kvutils/src/test/suite/scala/com/daml/ledger/participant/state/kvutils/KVUtilsTransactionSpec.scala [hunk 2]
```diff
@@ -82,6 +84,31 @@ class KVUtilsTransactionSpec extends AnyWordSpec with Matchers {
     val p0 = mkParticipantId(0)
     val p1 = mkParticipantId(1)
 
+    "add the key of an exercised contract as a submission input" in KVTest.runTestWithSimplePackage(
```
Comment (SamirTalwar-DA) [inline ledger/participant-state/kvutils/src/test/suite/scala/com/daml/ledger/participant/state/kvutils/KVUtilsTransactionSpec.scala:87]: I take it this test fails without the changes to `InputsAndEffects`?

Comment (fabiotudone-da) [reply ledger/participant-state/kvutils/src/test/suite/scala/com/daml/ledger/participant/state/kvutils/KVUtilsTransactionSpec.scala:87]: Yes.

File: ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/InputsAndEffects.scala (hunk 1 above)
Comment (fabiotudone-da) [reply ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/InputsAndEffects.scala:108]: Yes, key is filled if there is a key, not if it's by key.

**Review by SamirTalwar-DA — APPROVED** [review]

File: ledger/participant-state/kvutils/src/test/suite/scala/com/daml/ledger/participant/state/kvutils/KVUtilsTransactionSpec.scala [hunk 3]
```diff
@@ -355,5 +373,22 @@ class KVUtilsTransactionSpec extends AnyWordSpec with Matchers {
     }
   }
 
+  private def seed(i: Int) = {
```
Comment (miklos-da) [inline ledger/participant-state/kvutils/src/test/suite/scala/com/daml/ledger/participant/state/kvutils/KVUtilsTransactionSpec.scala:376]:
```suggestion
  private def seed(i: Int) =
```

File: ledger/participant-state/kvutils/src/test/suite/scala/com/daml/ledger/participant/state/kvutils/KVUtilsTransactionSpec.scala [hunk 4]
```diff
@@ -355,5 +373,22 @@ class KVUtilsTransactionSpec extends AnyWordSpec with Matchers {
...
+    crypto.Hash.hashPrivateKey(this.getClass.getName + i.toString)
+  }
```
Comment (miklos-da) [inline ledger/participant-state/kvutils/src/test/suite/scala/com/daml/ledger/participant/state/kvutils/KVUtilsTransactionSpec.scala:378]:
```suggestion
```

**Review by miklos-da — APPROVED** [review]

**Pull request merged**